lib/Models/WebMapTileServiceCatalogItem.ts

```diff
@@ -0,0 +1,688 @@
+    const styles: ReadonlyArray<CapabilitiesStyle> =
+      layer && layer.Style
+        ? Array.isArray(layer.Style)
+          ? layer.Style
+          : [layer.Style]
+        : [];
+    result.push({
+      layerName: layer?.Identifier,
+      styles: styles.map((style: CapabilitiesStyle) => {
+        let wmtsLegendUrl: CapabilitiesLegend | undefined = isReadOnlyArray(
+          style.LegendURL
+        )
+          ? style.LegendURL[0]
+          : style.LegendURL;
+        let legendUri, legendMimeType;
+        if (
+          wmtsLegendUrl &&
+          wmtsLegendUrl.OnlineResource &&
+          wmtsLegendUrl.OnlineResource["xlink:href"]
+        ) {
+          legendUri = new URI(
+            decodeURIComponent(wmtsLegendUrl.OnlineResource["xlink:href"])
+          );
+          legendMimeType = wmtsLegendUrl.Format;
+        }
+        const legend = !legendUri
+          ? undefined
+          : createStratumInstance(LegendTraits, {
+              url: legendUri.toString(),
+              urlMimeType: legendMimeType,
+              title: layer?.Identifier
+            });
+        return {
+          identifier: style.Identifier,
+          isDefault: style.isDefault,
+          abstract: style.Abstract,
+          legend: legend
+        };
+      })
+    });
+
+    return result;
+  }
+
+  @computed
+  get usableTileMatrixSets() {
+    const usableTileMatrixSets: { [key: string]: UsableTileMatrixSets } = {
+      "urn:ogc:def:wkss:OGC:1.0:GoogleMapsCompatible": {
+        identifiers: ["0"],
+        tileWidth: 256,
+        tileHeight: 256
+      }
+    };
+
+    const standardTilingScheme = new WebMercatorTilingScheme();
+
+    const matrixSets = this.capabilities.tileMatrixSets;
+    if (matrixSets === undefined) {
+      return;
+    }
+    for (let i = 0; i < matrixSets.length; i++) {
+      const matrixSet = matrixSets[i];
+      if (
+        matrixSet.SupportedCRS !== "urn:ogc:def:crs:EPSG::900913" &&
+        matrixSet.SupportedCRS !== "urn:ogc:def:crs:EPSG:6.18:3:3857" &&
+        matrixSet.SupportedCRS !== "urn:ogc:def:crs:EPSG:6.18.3:3857" && // found in esri wmts
+        matrixSet.SupportedCRS !== "urn:ogc:def:crs:EPSG::3857"
+      ) {
+        continue;
+      }
+      // Usable tile matrix sets must have a single 256x256 tile at the root.
+      const matrices = matrixSet.TileMatrix;
+      if (!isDefined(matrices) || matrices.length < 1) {
+        continue;
+      }
+
+      const levelZeroMatrix = matrices[0];
+      /* if (
+        (levelZeroMatrix.TileWidth | 0) !== 256 ||
+        (levelZeroMatrix.TileHeight | 0) !== 256 ||
+        (levelZeroMatrix.MatrixWidth | 0) !== 1 ||
+        (levelZeroMatrix.MatrixHeight | 0) !== 1
+      ) {
+        continue;
+      }
+
+      const levelZeroScaleDenominator = 559082264.0287178; // from WMTS 1.0.0 spec section E.4.
+      if (
+        Math.abs(levelZeroMatrix.ScaleDenominator - levelZeroScaleDenominator) >
+        1
+      ) {
+        continue;
+      } */
```

Comment:
@nf-s what do you think about this, is this part necessary, I changed `imageryProvider` to use provided `tileWidth` and `tileHeight`. Also, it can happen that the list of matrices doesn't contain levelZeroMatrix (it start from 10 for example)

Reply:
I don't much experience working with tile-sets like this (or the WMTS spec) - but this seems reasonable to me!
Let me have a look into it and I'll get back to you

Reply:
Yep I think you're right!